text/0003-placeholders-in-conditions.md

```diff
@@ -0,0 +1,82 @@
+- Start Date: 2023-06-15
+- Target Major Version: 2.x
+- Reference Issues: https://github.com/cedar-policy/cedar/issues/81
+- Implementation PR:
+
+# Summary
+
+__Support template placeholders in conditions__
+
+This RFC proposes to allow the `?principal` and `?resource` placeholders to appear in the `when`/`unless` conditions of a policy, not just in the policy scope.
+
+NB: This RFC was migrated from https://github.com/cedar-policy/cedar/issues/81 as our RFC process was introduced and formalized.
+
+# Basic example
+
+Example 1, from @WanderingStar
+
+```
+@id("Admin")
+permit(
+  principal == ?principal,
+  action,
+  resource)
+when {
+    // take any action in the scope that you're admin on
+    resource in ?resource
+
+    // allow access up the tree for navigation
+    || (action == Action::"Navigate" && ?resource in resource)
+};
+```
+
+Example 2
+
+```
+permit(principal, action, resource)
+when {
+    ?principal has currentAccessLevel
```

Comment:
I have a problem with this policy. `?principal` doesn't actually relate to `principal` in any way, yet this policy looks very much like we've written

```
permit(principal, action, resource)
when {
    principal has currentAccessLevel
        && principal.currentAccessLevel >= 2
}
```

This isn't at all what the policy actual does! It will permit and `principal` if the condition is true for whatever entity happens to instantiate `?principal`.

Reply:
Agree with john here. If we want to support this policy, what we actually want is arbitrary names for slots. ie:
```
permit(principal, action, resource) 
when {
  ?target has currentAccessLevel
};
```